### PR TITLE
Upgrade jetbrains plugin, kotlin

### PR DESCRIPTION
**Pull request by fmatosqg**
I was trying to make some code/feature changes, but got repeatedly stuck when downloading Idea. 

This resulted in being forced to bump intellij plugin version, which forced me to bump kotlin version. Good news is now it's not necessary to download any Idea versions if they won't be used for compiling or running.

There's a known problem when `./gradlew buildPlugin` which fails to compile. However if run twice it works. Unfortunately I can't reproduce the problem on master, because it's being impossible for me to download Idea, it always times out.

Compiling and dry-run are good for 2.4 preview 7 and 3.0 preview 8. There's a minor code change necessary though, I'm not sure whether merging this is a good idea, but you have far more experience in backwards compatibility than me.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -51,12 +51,11 @@ if (!hasProperty('StudioCompilePath')) {
 }
 
 intellij {
-    version 'IC-15.0.5' // this value is not important, as long as you can compile
-
     pluginName 'adb_idea'
     updateSinceUntilBuild false
+    intellij.instrumentCode false
```
Comment (fmatosqg) [inline build.gradle]: probably not necessary

File: src/main/java/com/developerphil/adbidea/ui/MyDeviceChooser.java [hunk 2]
```diff
@@ -112,7 +113,7 @@ public MyDeviceChooser(boolean multipleSelection,
 
     myFacet = facet;
     myFilter = filter;
-    myMinSdkVersion = AndroidModuleInfo.getInstance(facet).getRuntimeMinSdkVersion();
+    myMinSdkVersion = AndroidModuleModel.get(facet).getRuntimeMinSdkVersion();
```
Comment (pbreault) [inline src/main/java/com/developerphil/adbidea/ui/MyDeviceChooser.java]: What is the reason for this change?

File: build.gradle [hunk 3]
```diff
@@ -51,12 +51,11 @@ if (!hasProperty('StudioCompilePath')) {
...
 
-    intellij.alternativeIdePath = project.hasProperty("StudioRunPath") ? StudioRunPath : StudioCompilePath
+    intellij.localPath = StudioCompilePath
```
Comment (pbreault) [inline build.gradle]: I'd rather keep the `project.hasProperty("StudioRunPath") ? StudioRunPath : StudioCompilePath` here.
It allows me to run a different studio than the compile one. Same thing as targetSdk vs minSdk in android. It helps me to test for backward compatibility.

**Comment by pbreault** [comment]
Excellent! Thanks for that, the new gradle plugin is cleaner!
Just a couple of comments and I'll merge it.

File: build.gradle (hunk 3 above)
Comment (fmatosqg) [inline build.gradle]: sure to revert it, this change shouldn't have made it to the branch.

File: src/main/java/com/developerphil/adbidea/ui/MyDeviceChooser.java (hunk 2 above)
Comment (fmatosqg) [inline src/main/java/com/developerphil/adbidea/ui/MyDeviceChooser.java]: I didn't try running it with versions previous to 2.4 preview 7, apparently `AndroidModuleInfo.getInstance` doesn't exist anymore? I'll try checking with some other AS versions to be sure.

If that's the case it would be recommended to either wait for 2.4 stable release or do some backward compatibility magic.

Comment (pbreault) [inline src/main/java/com/developerphil/adbidea/ui/MyDeviceChooser.java]: the `AndroidModuleInfo.getInstance(facet)` method is present in studio 3.0, which has just reached beta1 so it should be released soon enough. 

The current version of the plugin (1.4.1) has already dropped support for studio 2.3 ( well, users of 2.3 are still able to download the latest support that was compatible with studio2.3, which is adb idea 1.4.0). 

I don't plan to backport anything to 2.3 so we can safely revert this line.

Usually I try to support the current stable version and the current beta but this time I pushed a bug fix that broke this compatibility (I didn't think that any feature would be added before the 3.0 release). 

Tl;DR let's revert this line

**Comment by fmatosqg** [comment]
I think keeping intellij.localPath is better because it skips the unnecessary IDEA download. Other than that I rolled back the assignment value.

I also added a new commit that fixes a problem I experienced with `./gradlew clean compileJava`, now it's successful in the first try. And removed the `compile.exclude module`, I faintly remember that was used to solve the problem with duplicated dependencies (from forced IDEA idea.jar and AS idea.jar). Duplication should not be a problem anymore, solved by using the new intellij plugin that skips the download.

**Comment by pbreault** [comment]
Looking good, thank you!
